src/domain/usecases/ValidateNameUseCase.ts

```diff
@@ -0,0 +1,18 @@
+import { Id } from "$/domain/entities/Ref";
+import { FutureData } from "$/domain/entities/generic/Future";
+import { DataSetRepository } from "$/domain/repositories/DataSetRepository";
+
+export class ValidateNameUseCase {
+    constructor(private dataSetRepository: DataSetRepository) {}
+
+    execute(options: ValidateNameOptions): FutureData<boolean> {
+        return this.dataSetRepository.getByName(options.name).map(dataSets => {
+            return dataSets.some(
+                dataSet =>
+                    dataSet.id !== options.dataSetId &&
+                    dataSet.name.toLowerCase() === options.name.toLowerCase()
+            );
+        });
+    }
+}
+export type ValidateNameOptions = { name: string; dataSetId: Id };
```

Comment:
name/dataSetId, both some use dataSet or none. I'd change all the nesm: ValidateNameUseCase -> ValidateDataSetNameUseCase.